src/ui/Facet/Facet.ts

```diff
@@ -1285,11 +1286,34 @@ export class Facet extends Component {
   protected facetValueHasChanged() {
     this.updateQueryStateModel();
     this.rebuildValueElements();
+    this.clearDependentFacetsIfNoSelectedValuesRemain();
     Defer.defer(() => {
       this.updateAppearanceDependingOnState();
     });
   }
 
+  private clearDependentFacetsIfNoSelectedValuesRemain() {
+    if (this.getSelectedValues().length) {
+      return;
+    }
+
+    this.getDependentFacets().forEach(facet => facet.reset());
+  }
+
+  private getDependentFacets() {
+    const fieldOption = this.options.field;
+    const facets = this.searchInterface.getComponents<Facet>(Facet.ID) || [];
```

Comment:
I added the `|| []` because in the regular test environment, `this.searchInterface.getComponents` returns `undefined`. Do you prefer to set it to return an empty array in the beforeEach?

Reply:
You could use underscore.js `forEach` method, it handles empty/null/undefined values properly without throwing errors.